lib/rage/code_loader.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require "zeitwerk"
+
+class Rage::CodeLoader
+  def initialize
+    @loader = Zeitwerk::Loader.new
+  end
+
+  def setup
+    autoload_path = "#{Rage.root}/app"
+    enable_reloading = Rage.env.development?
+    enable_eager_loading = Rage.env.production?
+
+    @loader.push_dir(autoload_path)
+    # The first level of directories in app directory won't be treated as modules
+    # e.g. app/controllers/pages_controller.rb will be linked to PagesController class
+    # instead of Controllers::PagesController
+    @loader.collapse("#{Rage.root}/app/*")
+    @loader.enable_reloading if enable_reloading
+    @loader.setup
+    @loader.eager_load if enable_eager_loading
+  end
+
+  def reload
+    @loader.reload
+    Rage.__router.reset_routes
+    load("#{Rage.root}/config/routes.rb")
+  end
+
+  def reloading_enabled?
+    @loader.reloading_enabled?
```

Comment:
If I'm getting this right, this method always returns `true` if `enable_reloading` has been called. This means some of the errors we purposely ignore on reloading will also be ignored on server startup. Which will lead to different behaviour in development and production.

Let's use a custom flag here.

Reply:
That's true. Will do!

Reply:
@rsamoilov wait... Do we even need this flag now? The application will try to draw routes on server startup (in setup.rb) so it will fail if, for example some controller action is missing. And during request the middleware will catch all exception and present it to the end user. I don't really see why we need the reloading flag now.